packages/telemetry/index.cjs

```diff
@@ -46,40 +77,84 @@ const logEvent = async (eventName, dev, settings) => {
 				repo = md5(settings.gitRepo);
 			}
 
-			if (settings.database) {
+			if (databaseName) {
+				database = databaseName;
+			} else if (settings.database) {
+				//legacy - remove this post migration
 				database = settings.database;
 			}
 
-			if (settings.credentials.filename) {
+			if (settings.credentials?.filename) {
 				demoDb = md5(settings.credentials.filename) === md5('needful_things.duckdb');
 			}
 		}
 
-		let directoryHash = md5(process.env.HOME);
-		let codespaces = process.env.CODESPACES === 'true';
+		let homeDirectory = undefined;
+		let codespaces = false;
+
+		if (process.env) {
+			const { HOME, CODESPACES } = process.env;
+			homeDirectory = HOME;
+			if (CODESPACES) {
+				codespaces = CODESPACES === 'true';
+			}
+		}
 
 		if (usageStats === 'yes') {
-			projectProfile = await getProfile();
+			const projectProfile = await getProfile();
 			var analytics = new Analytics(wK);
-			analytics.track({
+			const payload = {
 				anonymousId: projectProfile.anonymousId,
 				event: eventName,
 				properties: {
 					devMode: dev,
 					repoHash: repo,
 					database: database, // logs database type (postgres, snowflake, etc.)
```

Comment:
@hughess storing additional fields that are now available via USQL in hashed form.  Can be removed, unhashed etc, if need be.